README.md

```diff
@@ -219,6 +321,163 @@ module Maintenance
+        actual_ingredient_pairs = ShoppingListTask.enumerator_builder
+          .enumerator(context: context).to_a
+
+        assert_equal expected_ingredient_pairs, actual_ingredient_pairs
+      end
+    end
+
+    test '#count returns nil, as we cannot guess the recipe choice' do
+      assert_nil ShoppingListTask.count
+    end
+
+    private
+
+    def ingredient_pairs(recipe)
+      recipe.ingredients.map do |ingredient|
+        cursor = "#{recipe.id}:#{ingredient.id}"
+        [ingredient, cursor]
+      end
+    end
+  end
+end
+```
+
+Dependaing on its complexity, you may choose to test your Enumerator builder in
+isolation, in which case you can simplify the tests for your Task.
+
+```ruby
+# test/tasks/maintenance/ingredient_purge_task_test.rb
+module Maintenance
+  class IngredientPurgeTaskTest < ActiveSupport::TestCase
+    test '#process composts ingredients' do
+      ingredient = ingredients(:tomato)
+      ingredient.expects(:compost!)
+
+      IngredientPurgeTask.process(ingredient)
+    end
+
+    test '#enumerator_builder returns an ExpiredIngredientsEnumerator' do
+      assert_instance_of(
+        ExpiredIngredientsEnumerator,
+        IngredientPurgeTask.enumerator_builder,
+      )
+    end
+  end
+end
+```
+
+```ruby
+# test/models/expired_ingredients_enumerator_test.rb
+class ExpiredIngredientsEnumeratorTest < ActiveSupport::TestCase
+  setup do
+    [PantryAPI, FridgeAPI, FreezerAPI].each(&:enable_test_mode!)
+  end
+
+  teardown do
+    [PantryAPI, FridgeAPI, FreezerAPI].each(&:disable_test_mode!)
+  end
+
+  test '#enumerator enumerates expired ingredients, ignoring cursor' do
+    expirees = []
+
+    PantryAPI.stock(ingredients(:fresh_tomatoes))
+
+    ingredients(:green_potatoes).tap do |ingredient|
+      PantryAPI.stock(ingredient)
+      expirees << ingredient
+    end
+
+    ingredients(:curdled_milk).tap do |ingredient|
+      FridgeAPI.stock(ingredient)
+      expirees << ingredient
+    end
+
+    FridgeAPI.stock(ingredients(:leftover_pizza))
+
+    ingredients(
+      :entire_pack_of_strawberries_ruined_by_that_single_one_that_had_mold,
+    ).tap do |ingredient|
+      FridgeAPI.stock(ingredient)
+      expirees << ingredient
+    end
+
+    ingredients(:mysterious_container).tap do |ingredient|
+      FreezerAPI.stock(ingredient)
+      expirees << ingredient
+    end
+
+    FreezerAPI.stock(ingredients(:ice_cubes))
+
+    expected_pairs = expirees.map do |ingredient|
+      cursor = nil
+      [ingredient, cursor]
+    end
+
+    context = stub('context')
+    context.expects(:cursor).never
+
+    actual_pairs =
+      ExpiredIngredientsEnumerator.enumerator(context: context).to_a
+
+    assert_equal expected_pairs, actual_pairs
+  end
+
+  test '#enumerator is empty if no expired ingredients' do
+    PantryAPI.stock(ingredients(:tortillas))
+
+    assert_empty ExpiredIngredientsEnumerator.enumerator(context: context).to_a
+  end
+end
+```
+
+</details>
```

Comment:
This is a good idea! 👍  I think we might need to wrap the entire thing in a paragraph element to get the code blocks to work inside a collapsible:

```
<details><p>

{{code examples}}
</p></details>

Reply:
Yeah, I've run into that issue before, but I think the blank line is enough.

Indeed,


    <details><summary>Busted code snippet</summary>
    ```ruby
    puts ':('
    ```
    </details>

produces

<details><summary>Busted code snippet</summary>
```ruby
puts ':('
```
</details>

but in my experience all it needs to work is a blank line before it, as you've incidentally included in your example:

    <details><summary>Working code snippet</summary>

    ```ruby
    puts ':)'
    ```
    </details>
    
produces

<details><summary>Working code snippet</summary>

```ruby
puts ':)'
```
</details>

The `<p>` doesn't seem to be needed, unless I missed something.